src/main/java/org/matsim/run/LausitzScenario.java

```diff
@@ -225,4 +269,54 @@ public static void prepareCommercialTrafficConfig(Config config) {
 			);
 		}
 	}
+
+	private static void prepareVehicleTypesForEmissionAnalysis(Scenario scenario) {
+		for (VehicleType type : scenario.getVehicles().getVehicleTypes().values()) {
+			EngineInformation engineInformation = type.getEngineInformation();
+
+//				only set engine information if none are present
+			if (engineInformation.getAttributes().isEmpty()) {
+				switch (type.getId().toString()) {
+					case TransportMode.car -> {
+						VehicleUtils.setHbefaVehicleCategory(engineInformation, HbefaVehicleCategory.PASSENGER_CAR.toString());
+						VehicleUtils.setHbefaTechnology(engineInformation, AVERAGE);
+						VehicleUtils.setHbefaSizeClass(engineInformation, AVERAGE);
+						VehicleUtils.setHbefaEmissionsConcept(engineInformation, AVERAGE);
+					}
+					case TransportMode.ride -> {
+//							ignore ride, the mode routed on network, but then teleported
+						VehicleUtils.setHbefaVehicleCategory(engineInformation, HbefaVehicleCategory.NON_HBEFA_VEHICLE.toString());
+						VehicleUtils.setHbefaTechnology(engineInformation, AVERAGE);
+						VehicleUtils.setHbefaSizeClass(engineInformation, AVERAGE);
+						VehicleUtils.setHbefaEmissionsConcept(engineInformation, AVERAGE);
+					}
+					case FREIGHT -> {
+						VehicleUtils.setHbefaVehicleCategory(engineInformation, HbefaVehicleCategory.HEAVY_GOODS_VEHICLE.toString());
+						VehicleUtils.setHbefaTechnology(engineInformation, AVERAGE);
+						VehicleUtils.setHbefaSizeClass(engineInformation, AVERAGE);
+						VehicleUtils.setHbefaEmissionsConcept(engineInformation, AVERAGE);
+					}
+					case TransportMode.bike -> {
+//							ignore bikes
+						VehicleUtils.setHbefaVehicleCategory(engineInformation, HbefaVehicleCategory.NON_HBEFA_VEHICLE.toString());
+						VehicleUtils.setHbefaTechnology(engineInformation, AVERAGE);
+						VehicleUtils.setHbefaSizeClass(engineInformation, AVERAGE);
+						VehicleUtils.setHbefaEmissionsConcept(engineInformation, AVERAGE);
+					}
+					default -> throw new IllegalArgumentException("does not know how to handle vehicleType " + type.getId().toString());
+				}
+			}
+		}
+
+//			ignore all pt veh types
+		scenario.getTransitVehicles()
+			.getVehicleTypes()
+			.values().forEach(type -> VehicleUtils.setHbefaVehicleCategory(type.getEngineInformation(), HbefaVehicleCategory.NON_HBEFA_VEHICLE.toString()));
```

Comment:
A bit the same as for the links... should be IMO part of the input. Since this script here only sets average values, if no value is set, I do not know if the input has the engineInformation set correctly or not.

Maybe also make a bit more clear that you set here everything to average.